lib/plugins/aws/package/compile/functions/index.js

```diff
@@ -301,6 +301,71 @@ class AwsCompileFunctions {
         }
       }
 
+      const fileSystemConfigs = functionObject.fileSystemConfigs;
+
+      // TODO: Simplify these conditions
+      if (fileSystemConfigs && fileSystemConfigs.length !== 0) {
+        if (fileSystemConfigs.length > 1) {
+          // TODO: Add explanation - CloudFormation limitation
+          const errorMessage = 'You can specify at most 1 fileSystemConfig';
+          throw new this.serverless.classes.Error(errorMessage);
+        }
+
+        const fileSystemConfig = fileSystemConfigs[0];
+
+        if (!this.isArnRefGetAttOrImportValue(fileSystemConfig.arn)) {
+          const errorMessage = [
+            'fileSystemConfig.arn must be provided as an arn string,',
+            ' Ref, Fn::GetAtt or Fn::ImportValue',
+          ].join('');
+          throw new this.serverless.classes.Error(errorMessage);
+        }
+
+        const pathRegex = /^\/mnt\/[a-zA-Z0-9-_.]+$/;
+        const isMountPathValid = pathRegex.test(fileSystemConfig.localMountPath);
+
+        if (!isMountPathValid) {
+          const errorMessage = [
+            'fileSystemConfig.localMountPath must be provided as a string, ',
+            ` that satisfies the following pattern ${pathRegex}`,
+          ].join('');
+          throw new this.serverless.classes.Error(errorMessage);
+        }
+
+        // TODO: Implement a proper check
+        const shouldUpdateIamRole = true;
+        if (shouldUpdateIamRole) {
+          const iamRoleLambdaExecution = cfTemplate.Resources.IamRoleLambdaExecution;
+
+          const stmt = {
+            Effect: 'Allow',
+            Action: ['elasticfilesystem:ClientMount', 'elasticfilesystem:ClientWrite'],
+            Resource: [fileSystemConfig.arn],
+          };
+
+          // TODO: extract it as a util as it's used in multiple places?
+          // update the PolicyDocument statements (if default policy is used)
+          if (iamRoleLambdaExecution) {
+            iamRoleLambdaExecution.Properties.Policies[0].PolicyDocument.Statement = _.unionWith(
```

Comment:
I think `unionWith` is unjustified here (there's no chance in some other part we're adding exactly same statement. Let's just unconditonally push it